### PR TITLE
CR1: add styler.py script for review

### PR DESCRIPTION
**Pull request by j-faria**
This script provides a `styler` decorator that helps in making publication-ready figures, with the correct figure and font sizes.

---
This change is [Reviewable](https://reviewable.io/reviews/iastro-pt/codereviewclub/1)

### CODE REVIEW

File: styler.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import re
+import os
+
+this_folder = os.path.dirname(__file__)
+
+
+#### styler should also provide colors for consistency!
+## are these ok?
+## should actually be a dictionary with sane keys...
+colors = ['#4878cf', '#6acc65', '#d65f5f', '#b47cc7', '#c4ad66', '#77bedb']
```
Comment (DanielAndreasen) [inline styler.py]: Atom shows better colours, but maybe add some keys here?
`colors = {'blue': '#4878cf', 'green': ...}`

File: styler.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+default_figwidth = 3.543311946  # in inches = \hsize = 256.0748pt
+default_figheight_factor = 0.75
+
+def styler(func):
+  def inner(*args, **kwargs):
+    """
+    This function deals with the figure and axes
+    and makes them suitable to plot in the thesis
+    """
```
Comment (DanielAndreasen) [inline styler.py]: Which keywords can be used?

An example in the documentation would also be very helpful

File: styler.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    def __call__(self, x, pos=None):
+        # call the original ScalarFormatter
+        rv = ticker.ScalarFormatter.__call__(self, x, pos)
+        # check if we really use TeX
+        if plt.rcParams["text.usetex"]:
+            # if we have the string ^{- there is a negative exponent
+            # where the minus sign is replaced by the short hyphen
+            rv = re.sub(r'-', r'\mhyphen', rv)
+
+        rv = rv.replace('.0', '')
```
Comment (DanielAndreasen) [inline styler.py]: This might remove unwanted `.0`.

    if rv.endswith('.0'):
        rv = rv.replace('.0', '')

or something similar

**Review by DanielAndreasen — COMMENTED** [review]
Suggestions from CRC

File: styler.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
+# matplotlib parameters for publication plots
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+from matplotlib import gridspec
```
Comment (j-faria) [inline styler.py]: :+1:

File: styler.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+
+
+### for the tufte-style documents, these are some of the important lengths
+main_width = 4.22839541667  # in inches = \linewidth (in main text) = 304.44447pt
+margin_width = 1.952175 # in inches = \linewidth (in margin) =  140.5566pt
+full_width = 6.5046152778 # in inches = \linewidth (in fullwidth environment) = 468.3323pt
+
+### for the A&A article class
+AandA_width = 3.543311946  # in inches = \hsize = 256.0748pt
```
Comment (sousasag) [inline styler.py]: AandA width is 3.543311945991

Comment (DanielAndreasen) [reply styler.py]: How did you get this information?

Comment (j-faria) [reply styler.py]: Put `\the\hsize`in the Latex inside a figure and it prints 256.0748pt, then convert to inches

**Comment by jason-neal** [comment]
So is this all complete? 
Does this get merged now or should it stay untill the next one?

**Comment by j-faria** [comment]
I changed the script to a `CR1` folder and added a small readme file, as suggested in #2.
Corrected the points raised during the CRC meeting and added some documentation to the styler decorator. Ready to merge?

**Comment by DanielAndreasen** [comment]
Seems fine to me 👍

**Pull request merged**